test/acceptance/client.js

```diff
@@ -341,13 +408,65 @@ describe('The node.js OC client', function(){
         });
 
         it('should contain the error details', function(){
-          expect(error.toString()).to.match(/Error\: Server-side rendering failed: Error: (.*?) when connecting to http\:\/\/localhost\:1234 \(500\)/g);
+          var exp = getRegExpFromJson(expectedRequest),
+              expected = new RegExp('Error: Server-side rendering failed: request ' + exp + ' failed \\(Error: connect ECONNREFUSED(.*?)\\)');
+
+          expect(error.toString()).to.match(expected);
         });
       });
     });
 
     describe('when server-side rendering an existing component linked to a responsive registry', function(){
 
+      describe('when the component times-out', function(){
+
+        var error, result;
+
+        var expectedRequest = {
+          url: 'http://localhost:3030',
+          method: 'post',
+          headers: { 
+            accept: 'application/vnd.oc.unrendered+json',
+            'user-agent': 'oc-client-(.*?)'
+          },
+          timeout: 0.01,
+          json: true,
+          body: {
+            components: [{
+              name: 'errors-component',
+              parameters: {
+                errorType: 'timeout',
+                timeout: 1000
+              }
+            }]
+          }
+        };
+
+        before(function(done){
+          client.renderComponent('errors-component', {
+            parameters: { errorType: 'timeout', timeout: 1000 },
+            timeout: 0.01,
+            disableFailoverRendering: true
+          }, function(err, html){
+            error = err;
+            result = html;
+            done();
+          });
+        });
+
+        it('should contain a blank html response', function(){
+          expect(result).to.eql('');
+        });
+
+        it('should contain the error details', function(){
+
+          var exp = getRegExpFromJson(expectedRequest),
+              expected = new RegExp('Error: Server-side rendering failed: request ' + exp + ' failed \\(timeout\\)');
```

Comment:
@matteofigus is it worth including actual timeout value in the error message as well (for diagnostic purposes)